### PR TITLE
feature: add ms gate

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Add IonQ MS native gate

*Testing done:*

unit testing

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/circuits/angled_gate.py [hunk 1]
```diff
@@ -114,3 +115,124 @@ def __eq__(self, other):
...
+            ascii_symbols (Sequence[str]): ASCII string symbols for the gate. These are used when
+                printing a diagram of a circuit. The length must be the same as `qubit_count`, and
+                index ordering is expected to correlate with the target ordering on the instruction.
+                For instance, if a CNOT instruction has the control qubit on the first index and
+                target qubit on the second index, the ASCII symbols should have `["C", "X"]` to
+                correlate a symbol with that index.
+
+        Raises:
+            ValueError: If the `qubit_count` is less than 1, `ascii_symbols` are `None`, or
+                `ascii_symbols` length != `qubit_count`, or `angle` is `None`
```
Comment (speller26) [inline src/braket/circuits/angled_gate.py:148]:
```suggestion
            ValueError: If `qubit_count` is less than 1, `ascii_symbols` are `None`,
                `ascii_symbols` length != `qubit_count`, or `angle_1` or `angle_2` are `None`
```

File: src/braket/circuits/angled_gate.py [hunk 2]
```diff
@@ -114,3 +115,124 @@ def __eq__(self, other):
...
+        """Returns the adjoint of this gate as a singleton list.
+
+        Returns:
+            List[Gate]: A list containing the gate with negated angle.
+        """
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, DoubleAngledGate)
```
Comment (speller26) [inline src/braket/circuits/angled_gate.py:219]: Two double angle gates with the same parameters will not in general be the same
```suggestion
            isinstance(other, type(self))
```

File: src/braket/circuits/angled_gate.py [hunk 3]
```diff
@@ -114,3 +115,124 @@ def __eq__(self, other):
...
+        )
+
+
+@singledispatch
+def _angles_equal(angle_1, angle_2):
+    return math.isclose(angle_1, angle_2)
+
+
+@_angles_equal.register
+def _(angle_1: FreeParameterExpression, angle_2: FreeParameterExpression):
```
Comment (speller26) [inline src/braket/circuits/angled_gate.py:237]: The second parameter type doesn't need to be constrained

File: src/braket/circuits/gates.py [hunk 4]
```diff
@@ -2121,6 +2204,28 @@ def get_angle(self, **kwargs):
     return type(self)(angle=new_angle)
 
 
+def get_angles(self, **kwargs):
```
Comment (speller26) [inline src/braket/circuits/gates.py:2207]: Does this need to be public? Honestly, I'm not sure `get_angle` and `format_complex` need to be public either, but we need to keep it that way for backwards compatibility.

File: src/braket/circuits/angled_gate.py (hunk 2 above)
Comment (ajberdy) [reply src/braket/circuits/angled_gate.py:219]: This isn't a criteria for equality as much as a short-circuit to ensure the actual comparisons don't error out. That said, I did forget to check name

**Review by speller26 — APPROVED** [review]

**Pull request merged**